dask/tests/test_graph_manipulation.py

```diff
@@ -237,16 +238,34 @@ def test_clone(layers):
     ],
 )
 def test_blockwise_clone_with_literals(literal):
+    """https://github.com/dask/dask/issues/8978
+
+    clone() on the result of a dask.array.blockwise operation with a (iterable) literal
+    argument
+    """
     arr = da.ones(10, chunks=1)
+    blk = da.blockwise(lambda a, l: a, "i", arr, "i", literal, None)
+    cln = clone(blk)
+    assert_no_common_keys(blk, cln, layers=True)
+    da.utils.assert_eq(blk, cln)
 
-    def noop(arr, lit):
-        return arr
 
-    blk = da.blockwise(noop, "x", arr, "x", literal, None)
+@pytest.mark.skipif("not da or not zarr")
+def test_blockwise_clone_with_no_indices():
+    """https://github.com/dask/dask/issues/9621
 
-    cln = clone(blk)
+    clone() on a Blockwise layer on top of a dependency layer with no indices
+    """
+    blk = da.from_zarr(zarr.ones(10))
```

Comment:
It's unfortunate that we need to rely on questionable behavior in `from_array` to test this. It would be nice to "fix" this in `from_array` itself ("why are we using `None` for something that is not technically a literal?"). However, I agree with your decision to avoid opening that can of worms :)